apps/desktop/src/autofill/main/main-desktop-autotype.service.ts

```diff
@@ -1,33 +1,73 @@
+import { ipcMain, globalShortcut } from "electron";
+
 import { autotype } from "@bitwarden/desktop-napi";
+import { LogService } from "@bitwarden/logging";
 
-import { DesktopAutotypeService } from "../services/desktop-autotype.service";
+import { WindowMain } from "../../main/window.main";
 
 export class MainDesktopAutotypeService {
-  constructor(private desktopAutotypeService: DesktopAutotypeService) {}
+  keySequence: string = "Alt+CommandOrControl+I";
+
+  constructor(
+    private logService: LogService,
+    private windowMain: WindowMain,
+  ) {}
 
   init() {
-    this.desktopAutotypeService.autotypeEnabled$.subscribe((enabled) => {
-      if (enabled) {
+    ipcMain.on("autofill.configureAutotype", (event, data) => {
+      if (data.enabled === true && !globalShortcut.isRegistered(this.keySequence)) {
         this.enableAutotype();
-      } else {
+      } else if (data.enabled === false && globalShortcut.isRegistered(this.keySequence)) {
         this.disableAutotype();
       }
     });
+
+    ipcMain.on("autofill.completeAutotypeRequest", (event, data) => {
+      const { response } = data;
+
+      if (
+        response.username != undefined &&
+        response.username != null &&
+        response.username.length > 0 &&
+        response.password != undefined &&
+        response.password != null &&
+        response.password.length > 0
+      ) {
+        this.doAutotype(response.username, response.password);
+      }
```

Comment:
🎨 would it make sense to extract a helper function that does the undefined/null and length check and pass username and password to that? Maybe it'd be useful as a util function elsewhere?

```suggestion
      if (isNonEmptyString(response.username) && isNonEmptyString(response.password)  {
        this.doAutotype(response.username, response.password);
      }
```

Reply:
Good idea! Updated in https://github.com/bitwarden/clients/pull/15840/commits/ed68c2343e1bb5d605b1bc1990633024ca3aea40.

It's a bit of a bulky function name but I'd rather be precise. We could also look into editing the string type but I won't dig into that in this pr 😂